source/develop/release-management/features/network/allowExplicitDnsConfiguration.html.md

```diff
@@ -0,0 +1,132 @@
+
+### DB
+Database needs to be updated so it can accommodate dns configuration. Two places will be altered:
+
+* network_attachments table
+* network table
+
+both this tables must accommodate multiple DNS records. Naive solution would be to store them comma separated into single column. Better solution would be creating separate table for that. That's subject of further discussion.
+
+### REST
+
+As mentioned, you can specify DNS Configuration at three places. Corresponding REST areas will be altered. But first we need to add new element 'dns_configuration' as: 
+
+```
+<xs:element name="dns_configuration" type="DnsConfiguration"/>
+
+<xs:complexType name="DnsConfiguration">
+  <xs:complexContent>
+    <xs:extension base="BaseResource">
+      <xs:sequence>
+        <xs:element maxOccurs="1" minOccurs="0" name="ips" type="Ips"/>
+      </xs:sequence>
+    </xs:extension>
+  </xs:complexContent>
+</xs:complexType>
+```
+
+#### Creating new Host
+```
+POST /ovirt-engine/api/hosts HTTP/1.1
+Content-Type: application/xml
+
+<host>
+    <name>$NAME</name>
+    <address>$ADDRESS</address>
+    <port>$PORT</port>>
+    <root_password>$PASSWORD</root_password>
+    <cluster>
+        <name>$CLUSTERNAME</name>
+    </cluster>
+    <dns_configuration>
+        <ips>
+          <ip>
+            <address>8.8.8.8</address>
+          </ip>
+        </ips>
+      </dns_configuration>
+</host>
+```
+
+#### Updating Management Network
+only management network can be updated with dns configuration.
+
+```
+PUT /ovirt-engine/api/networks/{network:id} HTTP/1.1
+Content-Type: application/xml
+
+<network>
+  <dns_configuration>
+    <ips>
+      <ip>
+        <address>8.8.8.8</address>
+      </ip>
+    </ips>
+  </dns_configuration>
+</network>
+```
+
+#### Updating 'network attachment' of Management Network on specific Host 
+only network attachment of management network can be updated with dns configuration.
+
+```
+PUT ovirt-engine/api/hosts/{host:id}/networkattachments/{networkattachment:id} HTTP/1.1
+Content-type: application/xml
+
+<network_attachment>
+  <dns_configuration>
+    <ips>
+      <ip>
+        <address>8.8.8.8</address>
+      </ip>
+    </ips>
+  </dns_configuration>
+</network_attachment>
+
+```
+
+### GUI
+
+As mentioned, you can specify DNS Configuration at three places:
+
+#### Creating new Host
+![New Host Dialog with DNS Configuration](newHostDialogWithDnsConfiguration.png "New Host Dialog with DNS Configuration")
+
+#### Updating Management Network
+(note — when editing other network this option won't be available)
+![Edit Logical Network Dialog with DNS Configuration](editLogicalNetworkDialogWithDnsConfiguration.png "Edit Logical Network Dialog with DNS Configuration")
+
+#### Updating 'attachment' of Management Network on specific Host 
+![Editing Network Attachment Dialog with DNS configuration](editNetworkAttachmentDialogWithDnsConfiguration.png "Editing Network Attachment Dialog with DNS configuration")
```

Comment:
Please add a "Testing" section with:
* When no DNS is defined, the host predefined one remains configured. This is the case upon upgrade to ovirt-4.1 or fresh installation.
* When adding a host, its DNS is taken from the logical management network, overridden by network attachment.